agent_id/src/AgentID.cpp

```diff
@@ -26,20 +26,12 @@ uint8_t AgentID::getType() const
 
 bool AgentID::operator==(const AgentID& other) const
 {
-    if (_id.size() != other._id.size()) {
-        return false;
-    }
-    for (int i = 0; i < static_cast<int>(_id.size()); i++) {
-        if (_id[i] != other._id[i]) {
-            return false;
-        }
-    }
-    return true;
+    return (_id == other._id);
```

Comment:
_id is still a vector and does not implement ==,no?

Reply:
has been overloaded since c++11

Reply:
right!